htslib/hts.h

```diff
@@ -34,6 +34,7 @@ DEALINGS IN THE SOFTWARE.  */
 
 #include "hts_defs.h"
 #include "hts_log.h"
+#include "hts_expr.h"
```

Comment:
If the few declarations in _hts_expr.h_ are worthy of being in a separate header (FWIW +1 they are IMHO), might consider just forward declaring `struct hts_filter_t;` rather than #including the separate header here.

Reply:
Can do so, but it then also needs a similar modification to sam.h (or the include).  I'm not sure on what's best policy, but no strong views either way so will work on this.